packages/release-brancher/src/release-brancher.ts

```diff
@@ -0,0 +1,237 @@
+    if (existing) {
+      console.log(`branch ${this.branchName} already exists`);
+      return existing;
+    }
+
+    const sha = await this.getTargetSha(this.targetTag);
+    if (!sha) {
+      console.log(`couldn't find SHA for tag ${this.targetTag}`);
+      throw new Error(`couldn't find SHA for tag ${this.targetTag}`);
+    }
+
+    console.log(`creating branch ${this.branchName} as SHA ${sha}`);
+    const response = await this.octokit.git.createRef({
+      owner: this.upstreamOwner,
+      repo: this.upstreamRepo,
+      ref: `refs/heads/${this.branchName}`,
+      sha,
+    });
+    return response.data.ref;
+  }
+
+  async getFileContents(path: string): Promise<string | undefined> {
+    try {
+      const response = (await this.octokit.repos.getContent({
+        owner: this.upstreamOwner,
+        repo: this.upstreamRepo,
+        path,
+      })) as {data: {content: string}};
+      return Buffer.from(response.data.content, 'base64').toString('utf8');
+    } catch (e) {
+      if (e.status === 404) {
+        return undefined;
+      }
+      throw e;
+    }
+  }
+
+  updateReleasePleaseConfig(content: string): string | undefined {
+    const config = yaml.load(content) as ReleasePleaseConfig;
+    const branches = config.branches || [];
+    delete config.branches;
+
+    if (
+      branches.find(branch => {
+        return branch.branch === this.branchName;
+      })
+    ) {
+      // already found branch
+      return undefined;
+    }
+    const newConfig = yaml.load(content) as ReleasePleaseConfig;
+    const newBranchConfig: ReleasePleaseBranchConfig = {
+      ...config,
+      branch: this.branchName,
+    };
+    if (this.releaseType) {
+      newBranchConfig.releaseType = this.releaseType;
+    }
+    branches.push(newBranchConfig);
+    newConfig.branches = branches;
+    return yaml.dump(newConfig, {
+      noRefs: true,
+    });
+  }
+
+  updateSyncRepoSettings(content: string): string | undefined {
+    const config = yaml.load(content) as SyncRepoSettingsConfig;
+    const branches = config.branchProtectionRules || [];
+
+    if (branches.length === 0) {
+      // no configured branch protection - we cannot infer what to do
+      throw new Error('Cannot find exising branch protection rules: aborting');
+    }
+
+    if (
+      branches.find(branch => {
+        return branch.pattern === this.branchName;
+      })
+    ) {
+      // already found branch
+      return undefined;
+    }
+    // TODO: consider fetching the default branch name from the GitHub API
+    const found = branches[0];
+    const newRule = {
+      ...found,
+      pattern: this.branchName,
+    };
+    branches.push(newRule);
+    config.branchProtectionRules = branches;
+    return yaml.dump(config, {
+      noRefs: true,
+    });
+  }
+
+  /**
+   * Opens a pull request against the default branch with updated
+   * release-please and sync-repo-settings configurations. If an existing
```

Comment:
Oh my.  I'm starting to think we should update the `sync-repo-settings` bot to apply configuration right after changes land, instead of making folks wait for the cron job to do it's thing.  Not for this PR obviously, but can I get an issue to track?

Reply:
Opened #1517